lib/hash.js

```diff
@@ -0,0 +1,71 @@
+'use strict'
+
+const crypto = require('node:crypto')
+const fs = require('node:fs/promises')
+const path = require('node:path')
+const os = require('node:os')
+const { Glob } = require('glob')
+const fastq = require('fastq')
+
+async function generateFileHash (filePath) {
+  try {
+    const fileBuffer = await fs.readFile(filePath)
+    return `${crypto.createHash('md5').update(fileBuffer).digest('hex').slice(0, 16)}-`
+  } catch {
+    return ''
```

Comment:
This seems a bit sketchy to me -- why not let the error bubble out to the caller? If a file can't be read for a permissions error or some reason like that, I think that the user invoking the utility likely would want to know?

Reply:
This protects against the no-link files, which otherwise are not readable, and in these cases simply calling that route is more correct than doing something like crashing

I'm open to suggestions though